shipyard/models/spacecraft.py

```diff
@@ -126,3 +127,24 @@ def add_pplant(self, plant_type):
         self.cargo = self.cargo - new_pplant.tonnage
         self.fuel_two_weeks = new_pplant.fuel_two_weeks
 
+    def add_turret(self, turret):
```

Comment:
Should we check here if turret exceeds max hardpoints? Or just let things overrun and display in the red on GUI?

(legit question i haven't thought about it yet)

Reply:
To be honest, I hadn't actually considered hardpoints yet. Letting things overrun and letting an overarching handler might be more appropriate.
I'd reckon sticking with the latter option.